### PR TITLE
ENH: support rails 7

### PR DESCRIPTION
**Pull request by nastya-soroko**
Ref: #kraken-1043

### CODE REVIEW

File: db-query-matchers.gemspec [hunk 1]
```diff
@@ -15,10 +15,10 @@ Gem::Specification.new do |spec|
   spec.files         = Dir['lib/**/*.rb']
   spec.require_paths = ['lib']
 
-  spec.add_runtime_dependency 'activesupport', '>= 4.0', "< 7"
+  spec.add_runtime_dependency 'activesupport', '>= 4.0'
```
Comment (spieker) [inline db-query-matchers.gemspec:18]: @nastya-soroko have you tested that it still works? It seems, that someone intentionally has excluded Rails 7, do you know why?

File: lib/db_query_matchers/version.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
 # Defines the gem version.
 module DBQueryMatchers
-  VERSION = '0.11.0'
+  VERSION = '1.0.0'
```
Comment (spieker) [inline lib/db_query_matchers/version.rb]: Wouldn't this be more like a `0.11.1` release than a major jump?

**Review by spieker — CHANGES_REQUESTED** [review]
I think you should not add the `.gem` files to the repo.

File: db-query-matchers.gemspec (hunk 1 above)
Comment (nastya-soroko) [reply db-query-matchers.gemspec:18]: @spieker it is very tricky to test without a package due to the gem source, it was ok so far,  but I won't merge before the rails 7.0.0  test is entirely done. and from https://github.com/suitepad-gmbh/db-query-matchers/commit/7e710821317e573db8b2ad6897feaaa7f5fbfaef it was just historically with upper range, I don't see any obvious reasons for this yet.

File: lib/db_query_matchers/version.rb (hunk 2 above)
Comment (nastya-soroko) [reply lib/db_query_matchers/version.rb]: Yes, actually you are right, will change it

**Comment by spieker** [comment]
Still, I think the `.gem` files should not be part of the repository

**Comment by chiragrajk** [comment]
Is this PR still necessary? Since the original repo already has changes supporting rails 7.

**Comment by nastya-soroko** [comment]
@chiragrajk I don't even know why this fork exists, and how we maintain it. @spieker can you give little details?

**Comment by nastya-soroko** [comment]
@spieker @chiragrajk Kraken already uses the new version of this `db-query-matchers` (0.11.1). Please don't delete package 0.11.1. Once/if you decide to change the source of this gem please update in Kraken.

**Review by sehovicmirza — APPROVED** [review]

**Pull request merged**